mitodl/open-discussions
# add post detail display

#### What are the relevant tickets?

this is part of #27 

#### What's this PR do?

this PR basically implements the front-end UI for a 'detail' view of a `Post`. This includes a container and new route for that, a little bit of logic around when to fetch the relevant `Channel` and `Post` objects, and so on. I also updated the `Post` factory to reflect the latest thinking about what the API will look like.

#### How should this be manually tested?

From any thing which shows a list of posts you should be able to click on the `comments` link to get to a detail view for the post. If it is a text post, you should see something like this:

![textpost](https://user-images.githubusercontent.com/6207644/28588034-14154e00-7147-11e7-9566-ad350757264f.png)

If it's a URL post you should see basically the same thing, just without the blue text box.

You should be able to click around to the frontpage, to the subreddit, etc, and go to other posts (without a full-page refresh) with no problem.

Visiting a post URL directly (e.g. http://localhost:8063/channel/voluptatum/0.1173598155720883) should work as well.
